src/execution/aggregate_hashtable.cpp

```diff
@@ -308,10 +310,39 @@ idx_t GroupedAggregateHashTable::AddChunk(DataChunk &groups, Vector &group_hashe
 				}
 
 				distinct_addresses.Verify(new_group_count);
-
-				aggr.function.update(input_count == 0 ? nullptr : &payload.data[payload_idx], input_count,
-				                     distinct_addresses, new_group_count);
+				if (aggr.filter) {
```

Comment:
This seems duplicated from below. Perhaps better to move this into a function called "UpdateAggregate(...)" that is called from both places?